File: designpatterns/src/com/renho/structural/proxy/Proxy.java
package com.renho.structural.proxy;

public class Proxy implements Subject {

	private Subject realSubject;
	
	public Proxy() {
		this.realSubject = new RealSubject();
	}
	
	public Proxy(Subject realSubject) {
		this.realSubject = realSubject;
	}
	
	@Override
	public void request() {		
		this.realSubject.request();
	}

}

File: designpatternsCpp/src/structural/proxy/Proxy.cpp
/*
 * Proxy.cpp
 *
 *  Created on: 2015Äê4ÔÂ3ÈÕ
 *      Author: renho
 */

#include "Proxy.h"

Subject* subjecti;

int renho = 100;

Proxy::Proxy() {
}

Proxy::Proxy(Subject* subject) {
	subjecti = subject;
}

Proxy::~Proxy() {
}

void Proxy::request() {
	subjecti->request();
}
